src/evo/dmn_types.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_EVO_DMN_TYPES_H
+#define BITCOIN_EVO_DMN_TYPES_H
+
+#include <amount.h>
+#include <cassert>
+
+class CDeterministicMNType
+{
+public:
+    uint8_t index;
```

Comment:
may be change `uint8_t index` to `enum class MNType : uint8_t` ? 

Pros:
Firstly, switch inside `GetMnType()` and any similar code will show a compiler warning if any type is missing.
Secondly, instead "MnType::HighPerformance.index" you can write `MnType::HighPerformance`
Thirdly, you can remove .index from MnType
Cons:
Instead `(nType == MnType::HighPerformance.index)` need to write `nType == static_cast<int16_t>(MnType::HighPerformance)`

Reply:
see #5200